src/main/java/com/google/summit/translation/Translate.kt

```diff
@@ -388,11 +388,12 @@ class Translate(val file: String, private val tokens: TokenStream) : ApexParserB
         // This is an anonymous initializer. Define a method with a special name.
         val methodDecl =
           MethodDeclaration(
-            Identifier(MethodDeclaration.ANONYMOUS_INITIALIZER_NAME, SourceLocation.UNKNOWN),
+            id = Identifier(MethodDeclaration.ANONYMOUS_INITIALIZER_NAME, SourceLocation.UNKNOWN),
```

Comment:
Here, I started down the road of encoding magic method names. (The same thing could have happened for ctors.)

I like your idea to clearly indicate the special purpose with an explicit flag. Eventually I ought to go back and do a similar thing for anonymous initializers...

Reply:
I think either method is reasonable. Jorje uses `<init>` and `<clinit>` for constructors and `static` initialization blocks, respectively. In that case, getter properties like `isConstructor` and `isAnonymousInitialization` can still be added that make the interface more convenient.

I could even see this being preferable to flags because it is inherently mutually exclusive when there are multiple options, so maybe it's the constructor that should be changed to match anonymous blocks?

Reply:
+1 for making illegal states unrepresentable (... either with the `id` or through some other means)

Reply:
Decided I will do this as a separate PR since I need to edit anonymous init block code as well, and that is out of scope of this PR.

Reply:
Roger that